lib/sources.nix

```
@@ -229,6 +229,42 @@ let
 
   canCleanSource = src: src ? _isLibCleanSourceWith || !(pathHasContext (toString src));
 
+  /*
+    Get all files (recursively) that pass a filter function. Unlike other source
+    filtering functions, this will not include empty directories in the final
+    derivation.
+
```

```suggestion

This function always scans the entire directory tree. In large code bases, you can improve performance by filtering the second argument.
```

(assuming you're flipping the parameters as suggested below, and we fix `sourceLike` support)